### DISCUSSION TITLE
verifyReceipt() app_store sandbox returns empty {}

### DISCUSSION THREAD

**Discussion by mretcoyy** [imdhemy/laravel-in-app-purchases#170] [Q&A] [answered]
what is the possible problem with this app_store subscription verifyReceipt() returns {}

I test the base 64 and shared secret in revenue cat and it shows receipt.

**Comment by imdhemy** [comment]
Could you please provide a snippet of your code?

**Comment by mretcoyy** [comment]
use Imdhemy\GooglePlay\Subscriptions\SubscriptionPurchase;
use Imdhemy\Purchases\Facades\Subscription;


class UserSubscriptionController extends Controller
{

    public function validateSubscription(Request $request)
    {
        $validateUser = Validator::make($request->all(),
        [
            'purchase_token' => 'required',
            'provider' => 'required',
        ]);


        if($request->provider === "APPLE"){
            $receipt = $request->purchase_token;
            $receiptResponse = Subscription::appStore()->receiptData(
$receipt)->verifyRenewable();

            $receiptStatus = $receiptResponse->getStatus();

            if($receiptStatus->isValid()) {

                $latestReceiptInfo = $receiptStatus->getLatestReceiptInfo();
                return json_encode($latestReceiptInfo[0]);


            } else {

            }
        }
        else if($request->provider === "GOOGLE"){
            $itemId = "product_subs_annual";
            $purchaseToken = $request->purchase_token;
            $subscriptionReceipt = Subscription::googlePlay()->id($itemId)->
token($purchaseToken)->get();

            return json_encode($subscriptionReceipt);

        }
    }

**Comment by imdhemy** [comment, marked as answer]
The App store `LatestReceiptInfo` instance is not JSON serializable. The same applies to the `SupscriptionPurchase` from Google play.  

But still, you can get an array value from them.

``` diff
- return json_encode($latestReceiptInfo[0]);
+ return $latestReceiptInfo[0]->toArray() ;
```

```diff
- return json_encode($subscriptionReceipt);
+ return $subscriptionReceipt->toArray();
```

I deprecated the `verifyRenewable` [today](https://github.com/imdhemy/appstore-iap/releases/tag/1.4.0), so it would be better to replace it with the `verify` method as follows.

```diff
- $receiptResponse = Subscription::appStore()->receiptData($receipt)->verifyRenewable(); 
+ $receiptResponse = Subscription::appStore()->receiptData($receipt)->verify();
```
